pkg/services/vimmachine.go

```diff
@@ -127,32 +125,17 @@ func (v *VimMachineService) ReconcileNormal(machineCtx capvcontext.MachineContex
 		return false, err
 	}
 
-	// Convert the VM resource to unstructured data.
-	vmData, err := runtime.DefaultUnstructuredConverter.ToUnstructured(vm)
-	if err != nil {
-		return false, errors.Wrapf(err,
-			"failed to convert %s to unstructured data",
-			vm.GetObjectKind().GroupVersionKind().String())
-	}
-	vmObj := &unstructured.Unstructured{Object: vmData}
-	vmObj.SetGroupVersionKind(vm.GetObjectKind().GroupVersionKind())
-	vmObj.SetAPIVersion(vm.GetObjectKind().GroupVersionKind().GroupVersion().String())
-	vmObj.SetKind(vm.GetObjectKind().GroupVersionKind().Kind)
-
 	// Waits the VM's ready state.
-	if ok, err := v.waitReadyState(vimMachineCtx, vmObj); !ok {
-		if err != nil {
-			return false, errors.Wrapf(err, "unexpected error while reconciling ready state for %s", vimMachineCtx)
-		}
+	if ok := v.waitReadyState(vimMachineCtx, vm); !ok {
```

Comment:
Let's inline this function. Basically just check the field here directly
```
	if !vm.Status.Ready {
```
(I think the log in l.130 is okay we don't need another one)

Reply:
Sure, will update it.